Баженова Анна

bazhenova opened the pull request.

bazhenova committed:
First version

maxvipon commented on routes.js:
```diff
@@ -0,0 +1,13 @@
+	app.get('/', tasks.list);
+	app.post('/inserts', tasks.addTask);
+	app.post('/updates', tasks.updateTask);
+	app.post('/deletions', tasks.deleteTask);
```
Роуты совсем не по REST. Пересмотр лекцию и предложи исправление.

bazhenova replied:
Вместо post использовать delete и put (в некоторых местах)?

maxvipon commented on views/index.hbs:
```diff
@@ -0,0 +1,42 @@
+    </head>
+    <body>
+        <label class="refresh">
+            <img src="/animated.gif" alt="">
```
Для телефонов можно смело использовать SVG или анимацию на CSS

maxvipon commented on views/index.hbs:
```diff
@@ -0,0 +1,42 @@
+                {{#if tasks.length}}
+                    {{#each tasks}}
+                    <div>
+                        <div class="list__form">
```
По-хорошему должно быть формой

maxvipon commented on views/index.hbs:
```diff
@@ -0,0 +1,42 @@
+                            <div class="list__task">{{name}}</div>
+                            <input type="image" src="/trash.png" class="list__trash">
+                        </div>
+                        <div class="list__fix">
```
И это тоже

maxvipon commented on public/process.js:
```diff
@@ -0,0 +1,126 @@
+        var fix = this.parentElement;
+        var task = fix.parentElement;
+        var form = task.children[0];
+        form.style.display = 'block';
```
Почему не добавлением/удалением модификатора?

maxvipon commented on public/process.js:
```diff
@@ -0,0 +1,126 @@
+        var distanceX = Math.abs(end.pageX - start.pageX);
+        if (distanceX > 20) {
+            if (end.pageX < start.pageX) {
+                this.style.transform = 'translateX(-50px)';
```
Аналогичный вопрос

maxvipon commented on public/process.js:
```diff
@@ -0,0 +1,126 @@
+}
+
+function sendRequest(body, url, cb) {
+    var xhr = new XMLHttpRequest();
```
Слышала ли про `fetch`?

maxvipon commented on public/index.css:
```diff
@@ -0,0 +1,85 @@
+    margin: 0;
+}
+
+.refresh img
```
Там бэм, тут на бэм))

maxvipon commented on controllers/tasks.js:
```diff
@@ -0,0 +1,59 @@
+};
+
+exports.deleteTask = (req, res) => {
+    var index = parseInt(Object.keys(req.body)[0]);
```
Что может вернуть `parseInt` в случае некорректного входного параметра?

maxvipon commented on controllers/tasks.js:
```diff
@@ -0,0 +1,59 @@
+};
+
+exports.updateTask = (req, res) => {
+    var key = Object.keys(req.body)[0];
```
`key` может быть `undefined`

maxvipon commented on controllers/tasks.js:
```diff
@@ -0,0 +1,59 @@
+exports.updateTask = (req, res) => {
+    var key = Object.keys(req.body)[0];
+    var indexComma = key.indexOf(',');
+    var index = parseInt(key.substring(0, indexComma));
```
Тут лучше использовать `split`

maxvipon commented on controllers/tasks.js:
```diff
@@ -0,0 +1,59 @@
+    });
+};
+
+exports.refresh = (req, res) => {
```
А зачем в методе `refresh` добавление тасков?

bazhenova replied:
Переделала.

maxvipon replied:
Так, а зачем в методе `refresh` вообще всё это? Таски из разных клиентов добавляются/обновляются/удаляются явно соответствующими методами. Зачем здесь логика склейки? Вроде бы она должна быть на клиенте, нет?

maxvipon commented:
А есть heroku?

bazhenova committed:
Правки

- исправление багов
- использование fetch
- использование модификаторов
- использование форм

bazhenova commented:
@maxvipon Нет. Heroku я пока не освоила. В выходные приделаю реакт. Про REST коммент выше.

maxvipon commented on app.js:
```diff
@@ -0,0 +1,25 @@
+require('./routes')(app);
+
+console.log('Listening on port 8080');
+app.listen(8080);
```
_На будущее_: у метода `listen` есть второй аргумент `callback` – функция вызываемая после того, как сервер начал слушать порт или сокет. Сообщение в `console.log` должно быть в ней.

maxvipon commented on public/process.js:
```diff
@@ -0,0 +1,131 @@
+
+var globalStart;
+var globalEnd;
+document.addEventListener('touchstart', function(event) {
```
тут бы пустой строкой отбить, трудно читать

maxvipon commented on public/process.js:
```diff
@@ -0,0 +1,131 @@
+    event.stopPropagation();
+    globalStart = event.changedTouches[0];
+}, false);
+document.addEventListener('touchend', function(event) {
```
и тут

maxvipon commented on public/process.js:
```diff
@@ -0,0 +1,131 @@
+            return response.text();
+        })
+        .then(function(body) {
+            document.body.innerHTML = body;
```
Почем сразу не сделать `document.body.innerHTML = response.text()`?

maxvipon commented on public/process.js:
```diff
@@ -0,0 +1,131 @@
+            document.body.innerHTML = body;
+        })
+        .then(cb)
+        .catch(console.error);
```
Не сработает

maxvipon commented on public/process.js:
```diff
@@ -0,0 +1,131 @@
+    }, false);
+}
+
+function sendRequest(url, options, cb) {
```
Как-то некошерно использовать коллбэк, когда работаешь с промисами…

maxvipon replied:
Аня, расскажи, пожалуйста, что такое `Promise` и зачем они нужны?

maxvipon commented on routes.js:
```diff
@@ -0,0 +1,13 @@
+    app.get('/', tasks.list);
+    app.post('/inserts', tasks.addTask);
+    app.post('/updates', tasks.updateTask);
+    app.post('/deletions', tasks.deleteTask);
```
Продолжим:

> > Роуты совсем не по REST. Пересмотр лекцию и предложи исправление.
> 
> Вместо post использовать delete и put (в некоторых местах)?

Да. И ещё именование роутов. REST – это некая договоренность, стиль постоения API.
Нужно переделать.

maxvipon commented on views/index.hbs:
```diff
@@ -0,0 +1,42 @@
+                            <div class="list__task">{{name}}</div>
+                            <input type="image" src="/trash.png" class="list__trash">
+                        </form>
+                        <form class="list__fix">
```
`method="?" action="?"`

maxvipon commented on public/process.js:
```diff
@@ -0,0 +1,131 @@
+var saveButtons = document.getElementsByClassName('list__saving');
+saveButtons = Array.prototype.slice.call(saveButtons);
+for (var i = 0; i < saveButtons.length; i++) {
+    saveButtons[i].addEventListener('touchstart', function(event) {
```
Для того, чтобы не вешать листенер на каждый элемент – потому что это тупо дорого (вдруг у тебя в списке окажется 100500 задач) – существует [**делегация событий**](https://www.google.com/#q=mdn+event+delegation&gws_rd=ssl).

Нужно переделать.

maxvipon commented on views/index.hbs:
```diff
@@ -0,0 +1,42 @@
+    </head>
+    <body>
+        <label class="refresh">
+            <img src="/animated.gif" alt="" class="refresh__img">
```
> Для телефонов можно смело использовать SVG или анимацию на CSS

bazhenova replied:
Я помню про это.

bazhenova committed:
Переделанное частичное решение

bazhenova commented:
@maxvipon Доделала те правки, навешала события (без проблем не обошлось) Но. Теперь есть костыли, чтобы оно хоть как-то работало. Буду от них избавляться по мере приделывания реакта.

maxvipon commented:
Показывай :)

maxvipon commented on controllers/pages.js:
```diff
@@ -0,0 +1,13 @@
+}
+
+exports.index = (req, res) => {
+	res.render('index', {
```
табуляция съехала

maxvipon commented on routes.js:
```diff
@@ -0,0 +1,13 @@
+const pages = require('./controllers/pages');
+
+module.exports = function(app) {
+	app.get('/', pages.index);
```
табуляция съехала

maxvipon commented:
Вот теперь всё красиво, за исключением пары уехавших табуляций))
